bot/exts/utils/snekbox.py

```diff
@@ -141,10 +141,16 @@ async def callback(self, interaction: Interaction) -> None:
 
         Use a task calling snekbox, as run_job is blocking while it waits for edit/reaction on the message.
         """
-        # Call run_job in a task as blocking here would cause the interaction to be reported as failed by Discord's UI.
-        # Discord.py only ACKs the interaction after this callback finishes.
-        scheduling.create_task(self.snekbox_cog.run_job(self.job_name, self.ctx, self.version_to_switch_to, self.code))
-        await interaction.message.delete()
+        # Defer response here so that the Discord UI doesn't mark this interaction as failed if the job
+        # takes too long to run.
+        await interaction.response.defer()
+
+        with contextlib.suppress(NotFound):
+            # Suppress this delete to cover the case where a user re-runs code and very quickly clicks the button.
+            # The log arg on send_job will stop the actual job from running.
+            await interaction.message.delete()
```

Comment:
Doesn't hurt to have this but in my attempts the UI wouldn't let me click another button once one has been clicked.

Reply:
This was to cover when you edit the code and click the re-run emoji, then the Run in X button very quickly after. The timing window i pretty small though.